script/core/completion/completion.lua

```diff
@@ -2256,6 +2256,37 @@ local function tryluaDocOfFunction(doc, results, pad)
     }
 end
 
+---Checks for a lua symbol reference in comment and returns an async callback if found
+local function trySymbolReference(state, position, results)
+    local doc = getLuaDoc(state, position)
+    if not doc then
+        return
+    end
+
+    local line = doc.originalComment.text ---@type string
+    local col = select(2, guide.rowColOf(position)) - 2 ---@type integer
+
+    -- User will ask for completion at end of symbol name so we need to perform a reverse match to see if they are in a symbol reference
+    -- Matching in reverse allows the symbol to be of any length and we can still match all the way back to `](lua://` from right to left
+    local symbol = string.match(string.reverse(line), "%)?(.*)//:aul%(%]", #line - col)
```

Comment:
I am using `#line - col` to start the search right at the cursor to save a tiny bit of performance and allow multiple references within the same line.